src/main/java/com/puppycrawl/tools/checkstyle/xpath/AbstractNode.java

```diff
@@ -0,0 +1,361 @@
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Compares current object with specified for order.
+     * @return location
+     */
+    @Override
+    public final Location saveLocation() {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Compares current object with specified for position.
+     * @param nodeInfo another {@code NodeInfo} object
+     * @return constant from {@code AxisInfo} representing order of
+     *      current object to specified one
+     */
+    @Override
+    public final int comparePosition(NodeInfo nodeInfo) {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Returns head.
+     * @return head
+     */
+    @Override
+    public final Item head() {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Returns iterator.
+     * @return iterator
+     */
+    @Override
+    public final SequenceIterator iterate() {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Returns CharSequence string value.
+     * @return CharSequence string value
+     */
+    @Override
+    public final CharSequence getStringValueCS() {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Returns fingerprint.
+     * @return fingerprint
+     */
+    @Override
+    public final int getFingerprint() {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Returns display name.
+     * @return display name
+     */
+    @Override
+    public final String getDisplayName() {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Returns prefix.
+     * @return prefix
+     */
+    @Override
+    public final String getPrefix() {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Returns type of the schema.
+     * @return type of the schema
+     */
+    @Override
+    public final SchemaType getSchemaType() {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Returns AtomicSequence.
+     * @return AtomicSequence
+     */
+    @Override
+    public final AtomicSequence atomize() {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Generate id method.
+     * @param fastStringBuffer fastStringBuffer
+     */
+    @Override
+    public final void generateId(FastStringBuffer fastStringBuffer) {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Copy method.
+     * @param receiver receiver
+     * @param index index
+     * @param location location
+     */
+    @Override
+    public final void copy(Receiver receiver, int index, Location location) {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * To be implemented.
+     * @return UnsupportedOperationException exception
+     */
+    protected UnsupportedOperationException throwUnsupportedOperationException() {
```

Comment:
please make it private and make it clear in message that it this operation is not supported for all Nodes.
All child classes should have its own implementation.